packages/core/src/App/app.jsx

```diff
@@ -50,45 +52,24 @@ const App = ({ root_store }) => {
     }, []);
 
     React.useEffect(() => {
-        if (isMobile()) {
+        if (isTouchDevice() && isMobile()) {
             const el_landscape_blocker = document.getElementById('landscape_blocker');
 
-            const onFocus = () => {
-                /* Prevent from showing Landscape blocker UI when keyboard is visible */
-                el_landscape_blocker.classList.add('landscape-blocker--keyboard-visible');
-                root_store.ui.setIsNativepickerVisible(true);
-            };
-
-            const onFocusOut = e => {
-                if (e.target.classList.contains('dc-dropdown__display')) {
-                    // if the next target is a dropdown, keep native picker open
-                    return;
+            const handleResize = () => {
+                if (screen.availWidth <= screen.availHeight) {
+                    root_store.ui.onOrientationChange(false);
+                    el_landscape_blocker.classList.remove('landscape-blocker--visible');
+                } else {
+                    root_store.ui.onOrientationChange(true);
+                    el_landscape_blocker.classList.add('landscape-blocker--visible');
                 }
-                root_store.ui.setIsNativepickerVisible(false);
             };
 
-            const onTouchStart = () => {
-                if (document.activeElement.tagName !== 'INPUT') {
-                    el_landscape_blocker.classList.remove('landscape-blocker--keyboard-visible');
-                }
-            };
-            /**
-             * Adding `focus` and `focusout` event listeners to document here to detect for on-screen keyboard on mobile browsers
-             * and storing this value in UI-store to be used across the app stores.
-             *  - when document gets `focus` event - keyboard is visible
-             *  - when document gets `focusout` or `touchstart` event - keyboard is hidden
-             *  - note: the `touchstart` event comes after `focusout` and and we want to
-             *          remove `landscape-blocker--keyboard-visible` class as late as possible
-             * [TODO]: find an alternative solution to detect for on-screen keyboard
-             */
-            document.addEventListener('focus', onFocus, true);
-            document.addEventListener('focusout', onFocusOut, false);
-            document.addEventListener('touchstart', onTouchStart, true);
+            handleResize();
+            window.addEventListener('resize', debounce(handleResize, 400));
 
             return () => {
-                document.removeEventListener('focus', onFocus);
-                document.removeEventListener('focusout', onFocusOut);
-                document.removeEventListener('touchstart', onTouchStart);
+                window.removeEventListener('resize', debounce(handleResize, 400));
```

Comment:
Calling `debounce` will create a new function each time. We need to create the function outside so that we can refer to the same function in both `addEventListener` and `removeEventListener`
```
const debouncedHandleResize = debounce(handleResize, 400);
window.addEventListener('resize', debouncedHandleResize);

return () => {
    window.removeEventListener('resize', debouncedHandleResize);
}
```